<?xml version="1.0"?>
<launch>
    <param name="/use_sim_time" value="true"/>
    <arg name="base_global_planner" default="heterogeneous_formation_controller/hmfpcLocalPlannerROS"/>
    <arg name="base_local_planner" default="base_local_planner/TrajectoryPlannerROS dwa_local_planner/DWAPlannerROS "/>
    <arg name="local_planner" default="DWAPlannerROS" />
    <arg name="map_file" default="$(find heterogeneous_formation_controller)/config/map/map.yaml"/>
    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)">
        <param name="frame_id" value="odom" />
    </node>

    <node pkg="stage_ros" type="stageros" name="stageros" args="$(find heterogeneous_formation_controller)/config/stage/maze_carlike.world">
        <remap from="base_scan" to="/scan"/>
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find heterogeneous_formation_controller)/config/test-stage.rviz" />

    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
        <param name="base_global_planner" value="$(arg base_global_planner)"/>
        <!-- <param name="base_local_planner" value="$(arg base_local_planner)"/> -->
        <rosparam file="$(find mir_navigation)/config/Planner/LocalPlanner/$(arg local_planner).yaml" command="load" />

        <rosparam file="$(find heterogeneous_formation_controller)/config/move_base/planner.yaml" command="load"/>
        <rosparam file="$(find heterogeneous_formation_controller)/config/move_base/hmfpc_planner.yaml" command="load"/>

        <rosparam file="$(find heterogeneous_formation_controller)/config/move_base/costmap_common.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find heterogeneous_formation_controller)/config/move_base/costmap_common.yaml" command="load" ns="local_costmap" />

        <rosparam file="$(find heterogeneous_formation_controller)/config/move_base/costmap_local.yaml" command="load" ns="local_costmap" />

        <rosparam file="$(find heterogeneous_formation_controller)/config/move_base/costmap_global.yaml" command="load" ns="global_costmap" />
    </node>

</launch>